conans/client/tools/oss.py

```diff
@@ -288,6 +288,12 @@ def uname(options=None):
     @staticmethod
     def detect_windows_subsystem():
         from conans.client.tools.win import CYGWIN, MSYS2, MSYS, WSL
+        if OSInfo().is_linux:
+            try:
```

Comment:
I think it would be ok adding a comment linking to https://github.com/Microsoft/WSL/issues/423#issuecomment-221627364, as it is the best _official_ answer related to this hack.

Reply:
added